impeller/renderer/render_pass.cc

```diff
@@ -9,6 +9,9 @@ namespace impeller {
 RenderPass::RenderPass(std::weak_ptr<const Context> context,
                        const RenderTarget& target)
     : context_(std::move(context)),
+      sample_count_(target.GetSampleCount()),
```

Comment:
This could potentially lead to bugs since a RenderTarget's result to GetSampleCount isn't const.  A call to `RenderTarget::SetColorAttachment` can change the results of `GetSampleCount`.  Can we instead cache the result on the RenderTarget?  There we have the knowledge to know if it has become stale.

Reply:
The RenderTarget is a const property of the RenderPass, can only by accessed by a const reference, and is copied into the RenderPass on construction - this should be safe:

![image](https://github.com/flutter/engine/assets/8975114/05690595-118b-4fd9-91f1-63c7cf47accf)

`

Reply:
Okay, I just saw the copy.  This should be good.  Can you add a comment to before these cached values why it is safe to cache this values.  That at least gives a breadcrumb in case someone changes render_pass to a shared_ptr or something.  Something like "These values are cached for performance and are safe since RenderPass owns a const copy of the RenderTarget".

Reply:
Done!